providers/attio/parse.go

```diff
@@ -0,0 +1,24 @@
+// nolint
+package attio
+
+import (
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/spyzhov/ajson"
+)
+
+var dummyNextPageFunc = func(*ajson.Node) (string, error) {
+	return "", nil
+}
+
+// recordsWrapperFunc returns the records using the objectName dynamically.
+func recordsWrapperFunc(obj string) common.RecordsFunc {
+	return func(node *ajson.Node) ([]map[string]any, error) {
+		result, err := jsonquery.New(node).Array(obj, true)
```

Comment:
This method is similar to `common.GetRecordsUnderJSONPath()`. The only difference is that your version is ok with array being null as indicated by the `true` flag.

Maybe there should be 2 methods under common:
* GetRecordsUnderJSONPath -> which would be your version
* MustRecordsUnderJSONPath -> which is current impl GetRecordsUnderJSONPath.

Frankly, I am not sure that distinction is important!
I think before when I was refactoring, if key was not found that indicated that our code is at fault having wrong expectations in payload, thats about it... I have a feeling that you could change `GetRecordsUnderJSONPath` to have a `true` flag and use that method.